### PR TITLE
Add copy, assignment constructors to sndScifiHit

### PR DESCRIPTION
**Pull request by olantwin**
Also makes conversion from int explicit to avoid silent bugs.

Required for my solution to https://github.com/SND-LHC/sndsw/pull/211 memory leaks

### CODE REVIEW

**Comment by siilieva** [comment]
Does one really need to make the copy constructors public?
I am afraid copying lots of hits will make user's code slow.

**Comment by olantwin** [comment]
I don't see how not having copy constructors leads to faster code, especially when users use `sndScifiHit::Clone` instead, which might have been responsible for (part of) the memory leaks.

And copying (one way or another) is necessary when adding a hit to a TClonesArray or other container with contiguous memory.

Copying vs. moving etc. mostly is important for containers, not for their contents.

Happy to be convinced otherwise by benchmarks, examples or alternative solutions.

**Comment by siilieva** [comment]
I mean avoid making functions that do a copy, forced when there is no public copy ctor at hand, but rather passing the unique hit ID(the detID).
However, for the use case we have, it seems better to get a sub-sample of hits straight off(via copying) and not have to loop over all hits in later stages of an analysis(repeatedly for different stages). In this sense public copy ctor is what one needs.
In conclusion, I agree with these changes. Will proceed to the sndsw:master merge

**Comment by olantwin** [comment]
> I mean avoid making functions that do a copy, forced when there is no public copy ctor at hand, but rather passing the unique hit ID(the detID).

I agree, but for these kind of things, code review is probably the best solution, making sure things have sensible interfaces and avoid unnecessary copies. Disallowing copying is a very drastic solution, as copying has very legitimate uses.

PS: you might want to set your fork to also rebase PRs by default instead of making merge commits ;)

**Comment by siilieva** [comment]
I also noticed the `merge` - it is the 1st time someone else pushes here and I got spoiled with sndsw repo. :D
